onnxruntime/core/providers/dnnl/subgraph/dnnl_dequantizelinear.cc

```diff
@@ -43,18 +49,21 @@ void DnnlDequantizeLinear::CreatePrimitive(DnnlSubgraphPrimitive& sp, DnnlNode&
   if (axis < 0) {
     axis += x_dims;
   }
-  // If scale is a vector, add padding for broadcasting
-  if (!isScalar) {
-    Padd(&x_scale_md, static_cast<uint64_t>(axis + 1), x_dims);
+  // Prepare the scale to prevent broacasting errors
+  if (isScalar) {
+    // For scalar scale
+    Padd(&x_scale_md, x_dims, false);
+  } else {
+    // For N-D scale
+    Padd(&x_scale_md, static_cast<uint64_t>(axis) + 1, x_dims);
```

Comment:
Is there any reason why this change for scalar and non-scalar is needed?

Reply:
For some reason I was having an error on one of the test cases, fixed it by doing this

Reply:
cool